client/helpers.go

```diff
@@ -92,6 +91,10 @@ func changeTargetsDelegation(repo *tuf.Repo, c changelist.Change) error {
 		if err != nil {
 			return err
 		}
+		if data.IsWildDelegation(c.Scope()) {
```

Comment:
This is never getting exercised, which is weird because `cmd/notary/integration_test.go`'s `TestPurge` should exercise this bit of code.  

In `applyChangelist`, where we switch based on whether the scope is a delegation (L43) I think we also have to  change that line to be `isDel := data.IsDelegation(c.Scope()) || data.IsWildDelegation(c.Scope())`